crates/sui-types/src/crypto.rs

```diff
@@ -171,43 +230,146 @@ impl std::fmt::Debug for Signature {
     }
 }
 
-impl Signature {
-    pub fn new<T>(value: &T, secret: &dyn signature::Signer<Signature>) -> Self
-    where
-        T: Signable<Vec<u8>>,
-    {
-        let mut message = Vec::new();
-        value.write(&mut message);
-        secret.sign(&message)
+// 
+// BLS12381 Sui Signature port
```

Comment:
Remove later - is just here just to show as a POC how we can use different signature schemes.